internal/jimm/ssh/ssh.go

```diff
@@ -51,25 +51,50 @@ type SSHKeyManager interface {
 	VerifyPublicKey(ctx context.Context, claimUser string, publicKey []byte) (bool, error)
 }
 
+// SSHDialer provides a means to establish an SSH connection.
+type SSHDialer interface {
+	Dial(network string, addr string, config *gossh.ClientConfig) (*gossh.Client, error)
+}
+
+// BasicDialer is a wrapper around the default Go x/crypto/ssh
+// dialer for cases where no changes are needed.
+type BasicDialer struct{}
+
+func (d *BasicDialer) Dial(network string, addr string, config *gossh.ClientConfig) (*gossh.Client, error) {
+	return gossh.Dial(network, addr, config)
+}
+
+type SSHManagerParams struct {
+	IdentityManager IdentityManager
+	JujuManager     JujuManager
+	SSHKeyManager   SSHKeyManager
+	JWTFactory      *jujuauth.Factory
+	Dialer          SSHDialer
+}
+
 // NewSSHManager returns a new SSHManager that offers jimm functionality to the SSHJumpServer.
-func NewSSHManager(identityManager IdentityManager, jujuManager JujuManager, sshKeyManager SSHKeyManager, jwtFactory *jujuauth.Factory) (*sshManager, error) {
-	if identityManager == nil {
+func NewSSHManager(p SSHManagerParams) (*sshManager, error) {
+	if p.IdentityManager == nil {
```

Comment:
could all of this perhaps live in SSHManagerParams validate method?

Reply:
Yip, good idea.